app/controllers/v0/preferred_facilities_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module V0
+  class PreferredFacilitiesController < ApplicationController
+    def index
+      render(
+        json: current_user_preferred_facilities
+      )
+    end
+
+    def destroy
+      render(
+        json: current_user_preferred_facilities.find(params[:id]).destroy!
+      )
+    end
+
+    def create
+      preferred_facility = PreferredFacility.new(
+        params.require(:preferred_facility).permit(:facility_code).merge(
+          user: current_user
+        )
+      )
+
+      if preferred_facility.save
+        render(json: preferred_facility)
+      else
+        raise Common::Exceptions::ValidationErrors, preferred_facility
+      end
+    end
+
+    private
+
+    def current_user_preferred_facilities
+      current_user.account.preferred_facilities
```

Comment:
What should happen if the user doesn't have an account

Reply:
https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/user.rb#L48
an account model will be created if it doesn't already exist for a user